tfx/examples/bert/utils/bert_tokenizer_utils.py

```diff
@@ -0,0 +1,173 @@
+# Lint as: python2, python3
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Prepressing using tensorflow_text BertTokenizer"""
+
+import tensorflow as tf
+import tensorflow_text as text
+from tensorflow.python.eager.context import eager_mode
+import tensorflow_hub as hub
+
+_CLS = '[CLS]'
+_PAD = '[PAD]'
+_SEP = '[SEP]'
+
+class SpecialBertTokenizer():
+  """ Bert Tokenizer built ontop of tensorflow_text.BertTokenizer"""
+
+  def __init__(self, model_link):
+    self._model_link = model_link
+    self._find_special_tokens()
+
+  def _find_special_tokens(self):
+    """Find the special token ID's for [CLS] [PAD] [SEP]
+
+    Since each Bert model is trained on different vocaburary, it's important
+    to find the special token index pertaining to that model.
+    Since in Transform, tensorflow_hub.KerasLayer loads a symbolic tensor, turn
+    on eager mode to get the actual vocab_file location.
+    """
+
+    with eager_mode():
+      model = hub.KerasLayer(self._model_link)
+      vocab = model.resolved_object.vocab_file.asset_path.numpy()
+      self._do_lower_case = model.resolved_object.do_lower_case.numpy()
+      f = open(vocab, 'r')
+      self._sep_id = None
+      self._cls_id = None
+      self._pad_id = None
+      lines = f.read().split('\n')
+      for i, line in enumerate(lines):
+        if line == _PAD:
+          self._pad_id = i
+        elif line == _CLS:
+          self._cls_id = i
+        elif line == _SEP:
+          self._sep_id = i
+        if (self._pad_id is not None and
+            self._cls_id is not None and
+            self._sep_id is not None):
+          break
+
+  def tokenize_single_sentence(
+      self,
+      sequence,
+      max_len=128,
+      add_cls=True,
+      add_sep=True,
+      add_pad=True):
+    """Tokenize a single sentence according to the vocab used by the Bert model.
+
+    Add special tokens according to config.
+
+    Args:
+      sequence: Tensor of shape [batch_size, 1].
+      max_len: The number of tokens after padding and truncating.
+      add_cls: Whether to add CLS token at the front of each sequence.
+      add_sep: Whether to add SEP token at the end of each sequence.
+      add_pad: If not add_pad, would return ragged tensor instead of tensors
```

Comment:
I understand that the code between tokenize_single_sentence and tokenize two sentences is shared so perhaps the way to do this is to have a helper function that both call instead of having tokenize two sentences call the single sentence.

Reply:
The thinking behind this is: tokenize_single_sentece is the most fundamental operation introduced here.
Users can use it to accomplish different things, say output a ragged tensor and then do bucketized padding.
outputting different type seems fairly common.
maybe we can specify output type here?